data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,284 @@
+
+#Usage and help info for the script
+function help_usage()
+{
+        echo "script usage: $(basename $0) [-s Start Monitor for guest console] [-e End of Monitor for guest console ] [-h help]"
+}
+
+#Quit if there are less than one argument
+if [ $# -eq 0 ];then
+        help_usage
+        exit 1
+fi
+
+#PREPARATION
+##CONFIG PART
+START_MONITOR="0"
+END_MONITOR="0"
+LOG_DIR="/tmp/virt_logs_residence"
+MONITOR_LOCKED_FILE="/tmp/GUEST_CONSOLE_COLLECTOR_SEMAPHORE"
+timestamp="`date '+%Y%m%d%H%M%S'`"
+###CHECK EXISTED GUEST
+vm_guestnames_types="sles"
+get_vm_guestnames_inactive=`virsh list --inactive | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_inactive_array=$(echo -e ${get_vm_guestnames_inactive})
+get_vm_guestnames=`virsh list  --all | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_array=$(echo -e ${get_vm_guestnames})
+vmguest=""
+vmguest_failed="0"
+
+#Parse input arguments. -s or -e must have values
+while getopts "seh" OPTION; do
+  case "${OPTION}" in
+    s)
+      START_MONITOR="1"
+      ;;
+    e)
+      END_MONITOR="1"
+      ;;
+    h)
+      help_usage
+      exit 1
+      ;;
+    *)
+      help_usage
+      exit 1
+      ;;
+  esac
+done
+
+#FUNCTION PART
+function get_console()
+{
+
+local vmguest=$1
+
+expect -c "
+## -1 means never timeout
+set timeout -1
+
+##hide echo
+log_user 0
+spawn -noecho virsh console ${vmguest}
+
+#wait connection
+sleep 3
+send \"\r\n\r\n\r\n\"
+
+#condition expect
+expect {
+        \"*login:\" {
+                send \"root\r\"
+                exp_continue
+        }
+        \"*assword\" {
+                send \"novell\r\"
+                exp_continue
+        }
+        \"*:~ #\" {
+                send -- \"ip route get 1\r\"
+                exp_continue
+        }
+        \"*:~ #\" {
+                send -- \"exit\r\"
+                exp_continue                
+        }         
+}
+
+#submatch for output
+expect -re {dev.*\s([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})[^0-9]}
+
+if {![info exists expect_out(1,string)]} {
+        puts \"Match did not happen :(\"
+        exit 1
+}
+set output \$expect_out(buffer)
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+
+function open_console()
+{
+        #open guest console
+        for vmguest in ${vm_guestnames_state_running_array[@]};do
+                get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+        done 
+}
+
+function close_console()
+{
+        #close guest console
+        ##Kill parent PID from "basename $0" 
+        pid=`ps aux | grep "$(basename $0)" | grep -v "grep"| awk '{print $2}'`
+        kill -9 $pid >> /dev/null 2>&1
+        
+        ##Kill all associated "virsh cnsole" processes
+        for i in `seq 10`;do
+            ps aux | grep "virsh console" >> /dev/null 2>&1
+            if [ $? -eq 0 ];then
+                pid=`ps aux | grep "virsh console" | grep -v "grep"| awk '{print $2}'`
+                kill -9 $pid >> /dev/null 2>&1
```

Comment:
pid return may contain multiple pids. So you may need:
for pid in `ps axu | grep xxxxx | awk xxxx`;do
  kill -9  xxxx
done

And I do not think "seq 10" is good choice. You can use:

while pids return != ""  or similar

Reply:
> pid return may contain multiple pids. So you may need:
> for pid in `ps axu | grep xxxxx | awk xxxx`;do
> kill -9 xxxx
> done
sure, good idea. 

Also, had confirmed kill cmd support to kill with multiple pids.  
Such as
kill -9 XXX XXX XXX XXX... 
> 
> And I do not think "seq 10" is good choice. You can use:
> 
> while pids return != "" or similar

Cool, will confirm this

Reply:
had updated this